### ISSUE TITLE
exiftool does not read full metadata from WebP

### ISSUE THREAD

**Issue by ladd** [exiftool/exiftool#46]
Doing some testing with WebP files converted using ImageMagick, I note that exiftool will not read all the same metadata that imagemagick will.

```
$ exiftool imgh.webp 
ExifTool Version Number         : 11.97
File Name                       : imgh.webp
Directory                       : .
File Size                       : 2.1 MB
File Modification Date/Time     : 2020:04:24 16:02:30-07:00
File Access Date/Time           : 2020:04:28 09:52:38-07:00
File Inode Change Date/Time     : 2020:04:28 09:54:18-07:00
File Permissions                : rw-r--r--
File Type                       : WEBP
File Type Extension             : webp
MIME Type                       : image/webp
VP8 Version                     : 0 (bicubic reconstruction, normal loop)
Image Width                     : 3024
Horizontal Scale                : 0
Image Height                    : 4032
Vertical Scale                  : 0
Image Size                      : 3024x4032
Megapixels                      : 12.2

```

```
$ identify -verbose imgh.webp
Image: imgh.webp
  Format: WEBP (WebP Image Format)
  Mime type: image/webp
  Class: DirectClass
  Geometry: 3024x4032+0+0
  Resolution: 72x72
  Print size: 42x56
  Units: PixelsPerInch
  Colorspace: sRGB
  Type: TrueColor
  Base type: Undefined
  Endianess: Undefined
  Depth: 8-bit
  Channel depth:
    Red: 8-bit
    Green: 8-bit
    Blue: 8-bit
  Channel statistics:
    Pixels: 12192768
    Red:
      min: 0  (0)
      max: 255 (1)
      mean: 134.928 (0.529131)
      standard deviation: 66.8698 (0.262235)
      kurtosis: -1.13706
      skewness: -0.104765
      entropy: 0.989812
    Green:
      min: 0  (0)
      max: 255 (1)
      mean: 121.977 (0.478343)
      standard deviation: 62.1632 (0.243777)
      kurtosis: -0.827437
      skewness: 0.0323675
      entropy: 0.986513
    Blue:
      min: 0  (0)
      max: 255 (1)
      mean: 101.862 (0.399457)
      standard deviation: 65.3561 (0.256299)
      kurtosis: -0.800568
      skewness: 0.33422
      entropy: 0.980652
  Image statistics:
    Overall:
      min: 0  (0)
      max: 255 (1)
      mean: 119.589 (0.468977)
      standard deviation: 64.7964 (0.254103)
      kurtosis: -0.991201
      skewness: 0.0849857
      entropy: 0.985659
  Rendering intent: Perceptual
  Gamma: 0.454545
  Chromaticity:
    red primary: (0.64,0.33)
    green primary: (0.3,0.6)
    blue primary: (0.15,0.06)
    white point: (0.3127,0.329)
  Matte color: grey74
  Background color: white
  Border color: srgb(223,223,223)
  Transparent color: none
  Interlace: None
  Intensity: Undefined
  Compose: Over
  Page geometry: 3024x4032+0+0
  Dispose: Undefined
  Iterations: 0
  Compression: Undefined
  Orientation: TopLeft
  Convex hull: 0,0 3023,0 3023,4030 3022,4031 0,4031 0,0 
  Minimum bounding box: 3023,0 3023,4031 4.54747e-13,4031 2.0792e-13,0 
  Profiles:
    Profile-exif: 2320 bytes
  Properties:
    date:create: 2020-04-28T16:54:18+00:00
    date:modify: 2020-04-24T23:02:30+00:00
    exif:ApertureValue: 54823/32325
    exif:BrightnessValue: 59452/8295
    exif:ColorSpace: 65535
    exif:ComponentsConfiguration: 1, 2, 3, 0
    exif:DateTime: 2020:04:19 10:30:34
    exif:DateTimeDigitized: 2020:04:19 10:30:34
    exif:DateTimeOriginal: 2020:04:19 10:30:34
    exif:ExifOffset: 210
    exif:ExifVersion: 48, 50, 51, 49
    exif:ExposureBiasValue: 0/1
    exif:ExposureMode: 0
    exif:ExposureProgram: 2
    exif:ExposureTime: 1/312
    exif:Flash: 24
    exif:FlashPixVersion: 48, 49, 48, 48
    exif:FNumber: 9/5
    exif:FocalLength: 17/4
    exif:FocalLengthIn35mmFilm: 26
    exif:GPSAltitude: 277808/5225
    exif:GPSAltitudeRef: 0
    exif:GPSDateStamp: 2020:04:19
    exif:GPSDestBearing: 227042/3261
    exif:GPSDestBearingRef: M
    exif:GPSHPositioningError: 177928/8353
    exif:GPSImgDirection: 227042/3261
    exif:GPSImgDirectionRef: M
    exif:GPSInfo: 2038
    exif:GPSLatitude: 45/1, 32/1, 1499/100
    exif:GPSLatitudeRef: N
    exif:GPSLongitude: 122/1, 38/1, 2221/100
    exif:GPSLongitudeRef: W
    exif:GPSSpeed: 35602/14013
    exif:GPSSpeedRef: K
    exif:LensMake: Apple
    exif:LensModel: iPhone 11 Pro back triple camera 4.25mm f/1.8
    exif:LensSpecification: 807365/524263, 6/1, 9/5, 12/5
    exif:Make: Apple
    exif:MakerNote: 65, 112, 112, 108, 101, 32, 105, 79, 83, 0, 0, 1, 77, 77, 0, 26, 0, 1, 0, 9, 0, 0, 0, 1, 0, 0, 0, 11, 0, 2, 0, 7, 0, 0, 2, 46, 0, 0, 1, 76, 0, 3, 0, 7, 0, 0, 0, 104, 0, 0, 3, 122, 0, 4, 0, 9, 0, 0, 0, 1, 0, 0, 0, 1, 0, 5, 0, 9, 0, 0, 0, 1, 0, 0, 0, 177, 0, 6, 0, 9, 0, 0, 0, 1, 0, 0, 0, 175, 0, 7, 0, 9, 0, 0, 0, 1, 0, 0, 0, 1, 0, 8, 0, 10, 0, 0, 0, 3, 0, 0, 3, 226, 0, 12, 0, 10, 0, 0, 0, 2, 0, 0, 3, 250, 0, 13, 0, 9, 0, 0, 0, 1, 0, 0, 0, 21, 0, 14, 0, 9, 0, 0, 0, 1, 0, 0, 0, 4, 0, 16, 0, 9, 0, 0, 0, 1, 0, 0, 0, 1, 0, 17, 0, 2, 0, 0, 0, 37, 0, 0, 4, 10, 0, 20, 0, 9, 0, 0, 0, 1, 0, 0, 0, 10, 0, 23, 0, 9, 0, 0, 0, 1, 2, 80, 32, 0, 0, 25, 0, 9, 0, 0, 0, 1, 0, 0, 0, 2, 0, 26, 0, 2, 0, 0, 0, 6, 0, 0, 4, 48, 0, 31, 0, 9, 0, 0, 0, 1, 0, 0, 0, 0, 0, 32, 0, 2, 0, 0, 0, 37, 0, 0, 4, 54, 0, 33, 0, 10, 0, 0, 0, 1, 0, 0, 4, 92, 0, 35, 0, 9, 0, 0, 0, 2, 0, 0, 4, 100, 0, 37, 0, 9, 0, 0, 0, 1, 0, 0, 0, 134, 0, 38, 0, 9, 0, 0, 0, 1, 0, 0, 0, 3, 0, 39, 0, 10, 0, 0, 0, 1, 0, 0, 4, 108, 0, 40, 0, 9, 0, 0, 0, 1, 0, 0, 0, 1, 0, 43, 0, 2, 0, 0, 0, 37, 0, 0, 4, 116, 0, 0, 0, 0, 98, 112, 108, 105, 115, 116, 48, 48, 79, 17, 2, 0, 50, 1, 240, 0, 101, 1, 226, 1, 179, 0, 147, 0, 152, 0, 231, 0, 35, 1, 5, 1, 240, 1, 214, 1, 24, 2, 143, 2, 95, 2, 71, 2, 107, 1, 249, 0, 4, 1, 48, 1, 147, 0, 173, 0, 192, 0, 220, 0, 60, 1, 72, 1, 28, 2, 216, 1, 225, 1, 56, 2, 83, 2, 198, 1, 57, 1, 216, 0, 201, 0, 174, 0, 104, 0, 251, 0, 165, 0, 194, 0, 117, 1, 136, 1, 59, 2, 249, 1, 239, 1, 206, 1, 166, 1, 43, 1, 222, 0, 215, 0, 187, 0, 112, 0, 200, 0, 150, 0, 156, 0, 194, 0, 17, 1, 106, 1, 222, 1, 203, 1, 208, 1, 96, 1, 230, 0, 103, 0, 46, 1, 214, 0, 171, 0, 174, 0, 154, 0, 185, 0, 183, 0, 193, 0, 175, 0, 235, 0, 63, 1, 71, 1, 14, 1, 208, 0, 101, 0, 98, 0, 219, 0, 151, 0, 129, 0, 207, 0, 246, 0, 176, 0, 200, 0, 152, 0, 165, 0, 94, 0, 166, 0, 218, 0, 254, 0, 29, 1, 153, 0, 116, 0, 9, 1, 152, 0, 144, 0, 176, 0, 153, 0, 150, 0, 159, 0, 155, 0, 193, 0, 108, 0, 148, 0, 172, 0, 173, 0, 229, 0, 149, 0, 86, 0, 209, 0, 126, 0, 151, 0, 124, 0, 107, 0, 148, 0, 115, 0, 165, 0, 193, 0, 129, 0, 139, 0, 169, 0, 178, 0, 243, 0, 105, 0, 69, 0, 206, 0, 119, 0, 119, 0, 117, 0, 111, 0, 134, 0, 109, 0, 140, 0, 178, 0, 151, 0, 124, 0, 160, 0, 172, 0, 233, 0, 102, 0, 59, 0, 239, 0, 156, 0, 133, 0, 124, 0, 137, 0, 136, 0, 115, 0, 117, 0, 140, 0, 179, 0, 113, 0, 151, 0, 161, 0, 218, 0, 96, 0, 63, 0, 179, 0, 166, 0, 149, 0, 124, 0, 129, 0, 112, 0, 139, 0, 139, 0, 102, 0, 200, 0, 173, 0, 255, 0, 25, 1, 5, 1, 81, 0, 65, 0, 90, 0, 141, 0, 143, 0, 114, 0, 107, 0, 77, 0, 143, 0, 115, 0, 92, 0, 194, 0, 141, 0, 91, 0, 96, 0, 104, 0, 74, 0, 72, 0, 92, 0, 164, 0, 149, 0, 118, 0, 87, 0, 106, 0, 118, 0, 101, 0, 92, 0, 173, 0, 154, 0, 63, 0, 61, 0, 68, 0, 66, 0, 87, 0, 164, 0, 169, 0, 157, 0, 134, 0, 86, 0, 130, 0, 195, 0, 91, 0, 78, 0, 130, 0, 187, 0, 74, 0, 145, 0, 60, 2, 69, 1, 56, 1, 178, 0, 80, 0, 240, 0, 86, 0, 66, 0, 113, 0, 137, 0, 62, 0, 60, 0, 88, 0, 124, 0, 77, 0, 58, 1, 206, 3, 13, 3, 12, 2, 103, 0, 128, 0, 233, 0, 80, 0, 68, 0, 149, 0, 71, 0, 66, 0, 60, 0, 52, 0, 42, 0, 43, 0, 183, 1, 171, 3, 244, 2, 250, 2, 0, 8, 0, 0, 0, 0, 0, 0, 2, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, 12, 98, 112, 108, 105, 115, 116, 48, 48, 212, 1, 2, 3, 4, 5, 6, 7, 8, 85, 102, 108, 97, 103, 115, 85, 118, 97, 108, 117, 101, 89, 116, 105, 109, 101, 115, 99, 97, 108, 101, 85, 101, 112, 111, 99, 104, 16, 1, 19, 0, 4, 126, 194, 2, 65, 229, 238, 18, 59, 154, 202, 0, 16, 0, 8, 17, 23, 29, 39, 45, 47, 56, 61, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 9, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 63, 255, 255, 211, 219, 0, 2, 76, 10, 255, 255, 192, 35, 0, 0, 71, 251, 255, 255, 201, 139, 0, 0, 119, 2, 0, 0, 15, 191, 0, 0, 1, 0, 0, 0, 37, 39, 0, 0, 1, 0, 57, 51, 55, 49, 68, 57, 68, 53, 45, 67, 57, 54, 55, 45, 52, 66, 51, 68, 45, 57, 52, 54, 55, 45, 68, 65, 65, 51, 55, 70, 52, 70, 69, 57, 54, 54, 0, 0, 113, 57, 48, 48, 110, 0, 69, 66, 69, 54, 48, 53, 50, 52, 45, 68, 49, 56, 51, 45, 52, 65, 53, 48, 45, 56, 49, 69, 66, 45, 69, 49, 68, 52, 51, 67, 50, 53, 66, 66, 67, 66, 0, 0, 0, 0, 114, 53, 0, 0,
    exif:MeteringMode: 5
    exif:Model: iPhone 11 Pro
    exif:OffsetTime: -07:00
    exif:OffsetTimeDigitized: -07:00
    exif:OffsetTimeOriginal: -07:00
    exif:PhotographicSensitivity: 32
    exif:PixelXDimension: 4032
    exif:PixelYDimension: 3024
    exif:SceneCaptureType: 0
    exif:SceneType: 1
    exif:SensingMethod: 2
    exif:ShutterSpeedValue: 159911/19296
    exif:Software: 13.3.1
    exif:SubjectArea: 2002, 1505, 2213, 1324
    exif:SubSecTimeDigitized: 199
    exif:SubSecTimeOriginal: 199
    exif:WhiteBalance: 0
    exif:YCbCrPositioning: 1
    minimum-bounding-box:_p: 0,0
    minimum-bounding-box:_q: 3023,0
    minimum-bounding-box:_v: 3022,4031
    minimum-bounding-box:angle: 0
    minimum-bounding-box:area: 1.21857e+07
    minimum-bounding-box:height: 3023
    minimum-bounding-box:unrotate: -0
    minimum-bounding-box:width: 4031
    signature: ef114da02fdd8fd7e4e96c5c914fde7805520175774043201c16940a2238259c
    unknown: 2
  Artifacts:
    verbose: true
  Tainted: False
  Filesize: 2.11137MiB
  Number pixels: 12.1928M
  Pixels per second: 47.0556MP
  User time: 0.260u
  Elapsed time: 0:01.259
  Version: ImageMagick 7.0.10-7 Q16 x86_64 2020-04-20 https://imagemagick.org
```

[imgh.webp.zip](https://github.com/exiftool/exiftool/files/4547354/imgh.webp.zip)

**Comment by boardhead** [comment]
Thanks for this report, and for the sample image.  It seems that the WEBP EXIF suffers from the same problem as the HEIC EXIF -- nobody can agree on the proper header to use.  All previous samples I had start with a TIFF header, but this sample has an "Exif\0\0" header before the TIFF header.  I'll add support for this, and ExifTool 11.98 will be able to deal with files like this.

- Phil

**Comment by ladd** [comment]
Thanks for the fast reply!

**Comment by boardhead** [comment]
Fixed in 11.98
